docs/DesignDocs/CustomAttestation_V3.md

```diff
@@ -0,0 +1,197 @@
+Custom Attestation Data Formats for Open Enclave V3 Updates
+================
+
+As more custom attestation plugins are developed and more features are requested, new APIs are added to:
+
+1. Query registered attesters and verifiers.
+2. Select a format of the evidence that the registered attesters can generate.
+
+The following API was affected:
+
+1. Built-in SGX attester and verifier plugin
+
+This document describes the added and affected APIs as well as keeping
+the existing support for version 2 of the API.  This document assumes the user is familiar with
+the SGX attestation and [custom attestation](https://github.com/openenclave/openenclave/blob/master/docs/DesignDocs/CustomAttestation.md) with plugins.
+
+Motivation
+----------
+
+New APIs were added to help customers who would like to extend Open Enclave's current attestation structure with extra information while keep TEE-specific information within attester and verifier plugins.
+
+SGX report (referred as local attestation in Open Enclave) and SGX quote (referred as remote attestation in Open Enclave) generation takes different inputs and generates data of different structures. In order to simplify evidence generation in attestation applications, change was made to split built-in SGX attester into 2 attesters - one for SGX report and the other for SGX quote. Similar plugin splitting applies to built-in SGX verifier.
+
+Local and remote attestation is a TEE-specific feature. Specifying LOCAL or REMOTE flag when generating evidence can be confusing to many attesters. These flags were deprecated.
```

Comment:
Does this mean in V3, these flags are no longer defined, or they will be ignored by oe_get_evidence()?

Reply:
In V3, the flags parameter is still there but will be ignored by built-in SGX plugins.

---
In reply to: [387226785](https://github.com/openenclave/openenclave/pull/2621#discussion_r387226785) [](ancestors = 387226785)

Reply:
Could this mean that existing applications could be rebuilt with V3 API without modification, but would behave differently? If so the API would be broken at run-time instead of compile-time, and harder to debug.

Reply:
The flags definitions are deprecated in V3. The API will be broken at compile-time.

---
In reply to: [387830811](https://github.com/openenclave/openenclave/pull/2621#discussion_r387830811) [](ancestors = 387830811)